src/newViews/UserGroup/Memberships/index.tsx

```diff
@@ -0,0 +1,194 @@
+            title: _ts('usergroup', 'actionLabel'),
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: MembershipActionCell,
+            cellRendererParams: (passedUserId, data) => ({
+                member: data.member,
+                memberRole: data.role,
+                groupKey: data.group,
+                itemKey: passedUserId,
+                onEditClick: handleEditMemberClick,
+                onDeleteClick: memberDeleteTrigger,
+                editButtonTitle: _ts('usergroup', 'editMemberLabel'),
+                deleteButtonTitle: _ts('usergroup', 'deleteMemberLabel'),
+                deleteConfirmationMessage: _ts('usergroup', 'deleteMemberConfirmMessage'),
+                disabled: !canEdit || data.member === activeUserId,
+            }),
+        };
+
+        return ([
+            createStringColumn<Membership, number>(
+                'name',
+                _ts('usergroup', 'nameLabel'),
+                item => item.memberName,
+            ),
+            createStringColumn<Membership, number>(
+                'email',
+                _ts('usergroup', 'emailLabel'),
+                item => item.memberEmail,
+            ),
+            createDateColumn<Membership, number>(
+                'joinedAt',
+                _ts('usergroup', 'addedOnLabel'),
+                item => item.joinedAt,
+            ),
+            createStringColumn<Membership, number>(
+                'role',
+                _ts('usergroup', 'roleLabel'),
+                item => item.role,
+            ),
+            actionColumn,
+        ]);
+    }, [activeUserId, canEdit, handleEditMemberClick, memberDeleteTrigger]);
+
+    const memberOptions = useMemo(() => (
+        memberships?.results.map(d => ({
+            id: d.member,
+            displayName: d.memberName,
+        }))
+    ), [memberships?.results]);
+
+    return (
+        <Container
+            footerActions={(
+                <Pager
+                    activePage={activePage}
+                    itemsCount={memberships?.count ?? 0}
+                    onActivePageChange={setActivePage}
+                    maxItemsPerPage={MAX_ITEMS_PER_PAGE}
+                    itemsPerPageControlHidden
+                />
+            )}
+        >
+            {membershipsPending && <PendingMessage />}
+            <TableView
+                className={styles.expandedTable}
+                columns={membersColumns}
+                keySelector={membershipKeySelector}
+                data={memberships?.results}
+            />
+            {showAddUserModal && userGroup && (
+                <AddUserModal
+                    onModalClose={setUserModalHidden}
+                    group={userGroup}
+                    onUserAddSuccess={usersGetTrigger}
+                    userToEdit={userToEdit}
+                    memberOptions={memberOptions}
```

Comment:
Why pass memberOptions?

Reply:
The same form is used to get user details. By default the list needs to have the current user so we are sending the list.
We could send only the selected user, but its the same.

Reply:
Is that to preselect current user on the form?

Reply:
The prop name should be renamed.
Plus I noticed that the usage in AddUserModal is incorrect.